src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -51,12 +57,25 @@ public static class GetDownloadResultUtility
                                 return new DownloadResourceResult(DownloadResourceResultStatus.NotFound);
                             }
 
-                            return await GlobalPackagesFolderUtility.AddPackageAsync(
-                                identity,
-                                packageStream,
-                                settings,
-                                logger,
-                                token);
+                            if (cacheContext.NoCache)
+                            {
+                                return await AddPackageDirectAsync(
+                                    identity,
+                                    packageStream,
+                                    settings,
+                                    cacheContext.GeneratedTempFolder,
+                                    logger,
+                                    token);
+                            }
+                            else
+                            {
+                                return await GlobalPackagesFolderUtility.AddPackageAsync(
```

Comment:
It seems to be that almost all of the logic in `GlobalPackagesFolderUtility` could be shared with `AddPackageDirectAsync`, aside from the `SettingsUtility.GetGlobalPackagesFolder` call which, in the `AddPackageDirectAsync` case, is just the temp folder. Am I missing something? The indicator to me is that both are doing something very similar:

```csharp
var versionFolderPathContext = new VersionFolderPathContext(
    packageIdentity,
    PackagesDirectory,
    logger,
    packageSaveMode: PackageSaveMode.Nupkg | PackageSaveMode.Nuspec,
    xmlDocFileSaveMode: PackageExtractionBehavior.XmlDocFileSaveMode);

await PackageExtractor.InstallFromSourceAsync(
    stream => packageStream.CopyToAsync(stream, bufferSize: 8192, cancellationToken: token),
    versionFolderPathContext,
    token: token);

var defaultPackagePathResolver = new VersionFolderPathResolver(PackagesDirectory);
var path = defaultPackagePathResolver.GetPackageFilePath(packageIdentity.Id, packageIdentity.Version);
```